src/ut_kvp.c

```diff
@@ -610,6 +610,141 @@ uint32_t ut_kvp_getListCount( ut_kvp_instance_t *pInstance, const char *pszKey)
     return count;
 }
 
+unsigned char* ut_kvp_getDataBytes(ut_kvp_instance_t *pInstance, const char *pszKey, int *size)
+{
+    struct fy_node *node = NULL;
+    struct fy_node *root = NULL;
+    const char *byteString = NULL;
+    char zKey[UT_KVP_MAX_ELEMENT_SIZE];
+    char *token;
+    int byte_count = 0;
+    size_t buffer_size = 16; // Initial buffer size
+    unsigned char *output_bytes = (unsigned char *)malloc(buffer_size);
```

Comment:
try now to allocate in your variable definition, but do it where it's used the first time.

You've caused a leak in memory between 628 & 650 if any of your cases fail.

Then all the next error cases after that never free your memory